utils/filesystem.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef UTILS_FILESYSTEM_H
+#define UTILS_FILESYSTEM_H
+
+#include <stdbool.h>
+#include <sys/types.h>
+
+/*
+ * Check if the given path exist
+ * */
+bool pv_filesystem_path_exist(const char *path);
+/*
+ * check if the given path is a directory
+ * */
+bool pv_filesystem_path_is_directory(const char *path);
+/*
+ * Sync the given path. If path points to a file,
+ * then the parent folder is synced
+ * */
+void pv_filesystem_path_sync(const char *path);
+/*
+ * Path concatenation, the value is returned on the given buffer
+ * */
+void pv_filesystem_path_join(char *buf, int size, ...);
```

Comment:
maybe path_concat is better?

Reply:
`concat` is fine to me, I will change it

Reply:
Rename it 886df2e9a5b50522bfcf3c5b5d3984f78cc4a15d